vowpalwabbit/io/include/vw/io/io_adapter.h

```diff
@@ -104,7 +105,7 @@ std::unique_ptr<writer> open_compressed_stdout();
 std::unique_ptr<reader> open_stdin();
 std::unique_ptr<writer> open_stdout();
 
-using write_func_t = ssize_t (*)(void*, const char*, size_t);
+using write_func_t = std::function<ssize_t(void*, const char*, size_t)>;
```

Comment:
Not everything needs to become a `std::function` in my opinion. This function seems fine as it was.

Reply:
Changed back to function pointer